core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -174,7 +174,10 @@ public void close()
         exclusiveLock.interruptCurrentOwner();
 
         // if we can get the lock, attempt a clean shutdown; otherwise someone else will shutdown
-        tryWithLockUninterruptibly(() -> TRUE);
+        if (tryWithLockUninterruptibly(() -> TRUE).isEmpty()) {
+            // set the yield signal to stop the actively running driver as soon as possible
+            driverContext.getYieldSignal().yieldImmediatelyForTermination();
```

Comment:
Is there any downside of setting it unconditionally? (before `exclusiveLock.interruptCurrentOwner();`)